lib/Runtime/Library/TypedArray.cpp

```diff
@@ -1404,13 +1405,25 @@ namespace Js
 
         Assert(!(callInfo.Flags & CallFlags_New));
 
+        // 1. Let O be the this value.
+        // 2. If Type(O) is not Object, return undefined.
+        // 3. If O does not have a[[TypedArrayName]] internal slot, return undefined.
         if (args.Info.Count == 0 || !TypedArrayBase::Is(args[0]))
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedTypedArray);
+            if (scriptContext->GetConfig()->IsES6ToStringTagEnabled())
+            {
+                return library->GetUndefinedDisplayString();
```

Comment:
>GetUndefinedDisplayString [](start = 32, length = 25)

should we return GetUndefined() here instead of GetUndfinedDisplayString ? the spec says to return undefined.